src/Build/Evaluation/Conditionals/Scanner.cs

```diff
@@ -303,7 +312,12 @@ private string ParsePropertyOrItemMetadata()
         /// <summary>
         /// Scan for the end of the property expression
         /// </summary>
-        private static int ScanForPropertyExpressionEnd(string expression, int index)
+        /// <param name="expression">property expression to parse</param>
+        /// <param name="index">current index to start from</param>
+        /// <returns>A tuple result indicating whether the scan was successful. If successful, the index corresponds to the end of the property expression.
+        /// In case of scan failure, it is the error position index.
+        /// </returns>
+        private static (bool result, int index) ScanForPropertyExpressionEnd(string expression, int index)
```

Comment:
Might be cleaner to make it just return a bool and make index an out parameter.

Reply:
Indeed, fixed